tests/e2e/util.go

```diff
@@ -5147,24 +5147,38 @@ func getK8sMasterNodeIPWhereContainerLeaderIsRunning(ctx context.Context,
 
 	// Sorting the temporary file according to timestamp to find the latest container leader
 	// from the CSI pod replicas
-	cmd := "sort -k 2n leader.log | tail -1 | sed -n 's/.*podName://p' | tr -d '\n'"
-	framework.Logf("Invoking command '%v' on host %v", cmd,
-		k8sMasterIP)
-	result, err := sshExec(sshClientConfig, k8sMasterIP,
-		grepCmdForFindingCurrentLeader)
-	if err != nil || result.Code != 0 {
-		fssh.LogResult(result)
-		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
-			cmd, k8sMasterIP, err)
+	if containerName == syncerContainerName {
+		cmd := "sort -k 2n leader.log | tail -1 | sed -n 's/.*podName://p' | tr -d '\n'"
```

Comment:
only `cmd` looks different in `if` and `else` block, rest of parts like running cmd and setting `csiControllerPodName` is same in both places right, you can move them out of the loop